src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from datetime import datetime
+from typing import Any, Dict, List, Tuple, Union
+
+from django.db import transaction
+
+from sentry.issues.escalating import query_groups_past_counts
+from sentry.issues.escalating_issues_alg import generate_issue_forecast
+from sentry.models import Group, GroupStatus
+from sentry.models.groupforecast import GroupForecast
+from sentry.tasks.base import instrumented_task
+
+BATCH_SIZE = 1000
+GroupForecastTuple = Tuple[Group, List[int]]
+ParsedGroupCount = Dict[int, Dict[str, List[Union[str, int]]]]
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(
+    name="sentry.tasks.weekly_escalating_forecast.run_escalating_forecast",
+    queue="weekly_escalating_forecast",
+    max_retries=0,  # TODO: Increase this when the task is changed to run weekly
+)  # type: ignore
+def run_escalating_forecast() -> None:
+    archived_groups = list(
+        # TODO: Change this to Group.objects.filter(status=GroupStatus.IGNORED, substatus=GroupStatus.ARCHIVED_UNTIL_ESCALATING) once substatus is added
+        # TODO: Do not limit to project id = 1 and limit 10 once these topics are clarified
+        Group.objects.filter(
+            groupsnooze__until_escalating=True, status=GroupStatus.IGNORED, project__id=1
+        )[:10]
+    )
+    if not archived_groups:
+        return
+
+    response = query_groups_past_counts(archived_groups)
+    group_counts = parse_groups_past_counts(response)
+    group_forecast_list = get_forecast_per_group(archived_groups, group_counts)
+
+    with transaction.atomic():
+        # Delete old forecasts
+        GroupForecast.objects.filter(group__in=archived_groups).delete()
+
+        # Bulk create new forecasts
+        GroupForecast.objects.bulk_create(
+            [
+                GroupForecast(group=group, forecast=forecast)
+                for group, forecast in group_forecast_list
+            ],
+            batch_size=BATCH_SIZE,
+        )
```

Comment:
How many rows are likely to be involved here? We typically try to avoid transactions for extremely large updates, since it will lock those rows until the entire update is finished.

Reply:
We don't have a good idea of how many rows will be involved here, as we don't allow issues to be archived yet. We will be adding metrics to track this [here](https://getsentry.atlassian.net/browse/WOR-2769). Is there a better way to do this until we have a better idea of number of rows? Maybe batching `archived_groups` and running multiple transactions?

Reply:
Your idea to `create_or_update` might make sense as opposed to delete everything and create/re-create the rows in a tx.

Reply:
@wedamija which do you think would be better in this scenario - `create_or_update` or deletion and `bulk_create`?

Reply:
I think that once this is running on real data, you'll want to use `bulk_create`, but do it in batches of 100-1000.